checkout-core/src/main/java/com/adyen/checkout/core/exception/PermissionException.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by atef on 16/11/2022.
+ */
+
+package com.adyen.checkout.core.exception
+
+/**
+ *
+ * this exception indicates that the required runtime permission is not granted.
+ */
+class PermissionException(
+    errorMessage: String,
+    val requiredPermission: String
+) : CheckoutException(errorMessage)
+
+fun String.formatPermissionName(): String {
```

Comment:
I think that we should create a file called StringExtension and we add this extension there.

Reply:
I think we don't need that it's better to be in the same file related as it's very specific to `PermissionException`

Reply:
I agree with Onur, the `PermissionException` is a merchant facing class, and `String.formatPermissionName()` is an internal method that they don't need to know about or use. I'd rather have it somewhere else.

But I also don't like having it as a string extension, the reason is that whenever we have any string in the code base, we will be able to apply `formatPermissionName()` on it, while in fact this method 99% of the time will not be applicable.

Reply:
Sorry I didn't get back on that after talking with Arjen about this part we decided we won't be doing this as we won't show the permission name so I'll push an updated about this when the translations are finished.